### PR TITLE
Fix a bug where JWT on relative paths

### LINKED ISSUE #603: No host causes requests to set JWT
When making HTTP Request without Host, with `whitelistedDomains` set, it causes JWT Authorization to be set.

If we're using redirect to an S3 Assets Bucket, it causes the Authorization Header to be sent and AWS S3 will reject that, since it doesn't know it.

### Prerequisites

- [ x ] I have checked the [README documentation](https://github.com/auth0/angular2-jwt/blob/master/README.md).
- [ x ] I have checked the [Auth0 Community](https://community.auth0.com/) for related posts.
- [ x ] I have checked for related or duplicate [Issues](https://github.com/auth0/angular2-jwt
/issues) and [PRs](https://github.com/auth0/angular2-jwt
/pulls).
- [ x ] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [ x ] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [ x ] I am reporting this to the correct repository.

### Environment

Please provide the following:

@auth0/angular-jwt@2.1.0 on Angular 7

### Reproduction

When using `ngx-markdown` and ```[src]="/assets/foo/bar.md"```, the requests are intercepted by the JWT Interceptor but since ```requestUrl.host``` is null, cause the JWT Authorization header to be set.
Since we're using AWS ALB to redirect that to another sub-domain (S3 Backed), when Authorization is set, causes the requests to fail (since the Authorization isn't AWS Compatible)

### PR DESCRIPTION
**Pull request by driverpt**
Fixes #603

### CODE REVIEW

**Comment by driverpt** [comment]
Any prediction when this is going to be merged & released ?

**Comment by crooksey** [comment]
Are you aware this a duplicate of pull request #542 ?

**Comment by driverpt** [comment]
How is this a duplicate of #542 ?
image

**Comment by crooksey** [comment]
Sorry, commented on wrong issue.

File: src/jwt.interceptor.ts [hunk 1]
```diff
@@ -40,15 +40,15 @@ export class JwtInterceptor implements HttpInterceptor {
 
   isWhitelistedDomain(request: HttpRequest<any>): boolean {
     const requestUrl: any = parse(request.url, false, true);
-
+    const requestHost: string = requestUrl.host || (location && location.host)
```
Comment (davidpatrick) [inline src/jwt.interceptor.ts]: In order to not block potential universal support, can we do a `typeof` check here on location?

Comment (driverpt) [reply src/jwt.interceptor.ts]: image

```
typeof location === 'object'
```

???

Comment (driverpt) [reply src/jwt.interceptor.ts]: Done

**Comment by driverpt** [comment]
@stevehobbsdev , that is the correct behaviour. If you're redirecting to an S3 Bucket, since the initial request has no Host, it will always add the JWT Headers, which will result in 403 Errors from AWS.

**Pull request merged**